config/v1/types_console.go

```diff
@@ -26,6 +29,26 @@ type Console struct {
 type ConsoleSpec struct {
 	// +optional
 	Authentication ConsoleAuthentication `json:"authentication"`
+	// customization is used to optionally provide a small set of
+	// customization options to the web console.
+	// +optional
+	Customization ConsoleCustomization `json:"customization,omitempty"`
+}
+
+// ConsoleCustomization defines a list of optional configuration for the console.
+type ConsoleCustomization struct {
+	// customURL contains URL and certificate for replacing the default console URL.
+	// +optional
+	CustomURL `json:customURL,omitempty`
+}
+
+// CustomURL replaces the default console URL under which console will be available.
+type CustomURL struct {
+	// host is the desired custom URL under which console will be available.
+	Host string `json:host:omitempty`
+	// servingCert references to Secret in the openshift-config namespace that
+	// contains custom certificate and key.
+	ServingCert corev1.SecretReference `json:servingCert:omitempty`
```

Comment:
This probably doesn't need to be required. If you pick a hostname that uses the default routing suffix, you shouldn't need to specify a certificate and key.

Reply:
Makes sense 👍

Reply:
Should we document that that's a reasonable thing to do?